File: src/pocketcloud/cloud/event/impl/template/TemplateEvent.php
<?php

namespace pocketcloud\cloud\event\impl\template;

use pocketcloud\cloud\event\Event;
use pocketcloud\cloud\template\Template;

abstract class TemplateEvent extends Event {

    public function __construct(private readonly Template $template) {}

    public function getTemplate(): Template {
        return $this->template;
    }
}

File: src/main/java/de/pocketcloud/cloud/event/impl/template/TemplateEvent.java
package de.pocketcloud.cloud.event.impl.template;

import de.pocketcloud.cloud.event.Event;
import de.pocketcloud.cloud.template.Template;
import lombok.Getter;

public abstract class TemplateEvent extends Event {

    @Getter
    private final Template template;

    public TemplateEvent(Template template) {
        this.template = template;
    }
}